apps/air-discount-scheme/backend/src/app/modules/user/user.service.ts

```diff
@@ -82,4 +92,15 @@ export class UserService {
   async getUserInfoByNationalId(nationalId: string): Promise<User | null> {
     return this.getUserByNationalId<User>(nationalId, User)
   }
+
+  async getMultipleUsersByNationalIdArray(ids: string[]): Promise<User[]> {
+    const allUsers = ids.map(
+      async (nationalId) =>
+        (this.getUserInfoByNationalId(nationalId) as unknown) as User,
+    )
+
+    const result = (await Promise.all(allUsers)) as User[]
```

Comment:
Are all of these type casts necessary/correct? You need to handle the nulls somewhere.